### PR TITLE
feat(KvTabs): Accessible components for implementing a tab UI pattern

### PR DESCRIPTION
**Pull request by ryan-ludwig**
VUE-594

### CODE REVIEW

**Comment by ryan-ludwig** [comment]
Any feedback on how you imagine composing these tabs would be great. Is the slot syntax working for you? Does the provide/inject pattern make sense to you here?

File: @kiva/kv-components/vue/KvTab.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			const { navItems, selectedIndex } = this.$KvTabContext;
+			return navItems[selectedIndex]?.for === this.for;
+		},
+		index() {
+			const { navItems } = this.$KvTabContext;
+			return navItems?.findIndex((navItem) => navItem.for === this.for);
+		},
+	},
+	mounted() {
+		this.$KvTabContext.navItems.push(this);
```
Comment (ryan-ludwig) [inline @kiva/kv-components/vue/KvTab.vue:47]: This felt a little funny at first, but seems to work OK. We could also populate the navItems array inside KvTabs.vue by checking `$slots.tabNav` if this seems too weird.

Comment (mcstover) [reply @kiva/kv-components/vue/KvTab.vue:47]: I think this seems ok for now. My initial concern was just the extraneous data associated with each instance of a KvTab. This is a fairly light component...In other more complex cases I would def recommend we inject only what is needed.

**Review by mcstover — APPROVED** [review]

**Comment by emuvente** [comment]
The alternatives to provide/inject that come to mind would involve more complex code, so this is fine by me! I was initially concerned that it wouldn't work if you had multiple KvTabs on the page, but then I saw the story with multiple KvTabs working just fine. This looks great!

**Review by emuvente — APPROVED** [review]

**Comment by mcstover** [comment]
👍

**Review by BoulderBrains — APPROVED** [review]
The underline animation growing and shrinking when transition between tabs looks really nice.

**Review by mlibazisi — APPROVED** [review]
👍

**Pull request merged**